projects/optic/src/commands/capture/capture.ts

```diff
@@ -197,43 +202,71 @@ const getCaptureAction =
     }
 
     // make requests
-    const concurrency = captureConfig.config?.request_concurrency || 5;
-    const requests = makeRequests(
-      captureConfig.requests,
-      proxyUrl,
-      concurrency
-    );
-
     const captures = new GroupedCaptures(
       trafficDirectory,
       getEndpointsFromSpec(spec.jsonLike)
     );
-    const harEntries = HarEntries.fromProxyInteractions(proxyInteractions);
 
-    await Promise.all(requests)
-      .catch((error) => {
-        console.error(error);
-      })
-      .finally(() => {
-        // stop the proxy
-        sourcesController.abort();
-        // stop the app server
-        if (!options.serverOverride && captureConfig.server.command) {
-          process.kill(-app.pid!);
-        }
-      });
+    let requestsPromise: Promise<any> = Promise.resolve();
+    let reqCmdPromise: Promise<void> = Promise.resolve();
+    try {
+      if (captureConfig.requests) {
+        const requests = makeRequests(
+          captureConfig.requests,
+          proxyUrl,
+          captureConfig.config?.request_concurrency || 5
+        );
+        requestsPromise = Promise.all(requests);
+      }
+
+      if (captureConfig.requests_command) {
+        const cmd = commandSplitter(captureConfig.requests_command.command);
+        const proxyVar =
+          captureConfig.requests_command.proxy_variable || 'OPTIC_PROXY';
+        process.env[proxyVar] = proxyUrl;
+        const reqCmd = spawn(cmd.cmd, cmd.args, {
+          detached: true,
+          shell: true,
+        });
+
+        reqCmdPromise = new Promise((resolve, reject) => {
+          reqCmd.on('exit', (code) => {
+            if (code === 0) {
+              resolve();
+            } else {
+              reject();
+            }
+          });
+        });
+        reqCmd.stderr.on('data', (data) => {
+          logger.error(data.toString());
+        });
+      }
+    } catch (error) {
+      logger.error(error);
```

Comment:
I'm not too sure on this, but maybe we should throw the error here so that we bail out of the rest of the command.

Alternatively, we can use `await Promise.allSettled` which will wait for all of our requests to finish and we can disregard any requests that failed. What do you think?

Reply:
Thinking about it, we might want to:
- Collect the errors that are thrown from here
- log them out as a warning (e.g. `some requests failed, etc`) - but we should still be able to continue

Failed requests can be inconsistent (and there might be cases you want to fail on them) but for the most part I think we can continue

Reply:
yeah, i do think its correct to continue. since the command can be anything, like a script that runs multiple commands, i think its possible that we may be able to salvage some of the requests. or at least, that was my thinking at the time.

Reply:
@niclim were you thinking something like this, https://github.com/opticdev/optic/pull/2085/commits/4e54cb0d47a3d84a1fc98228818ec57e21bbc9b3?

Reply:
Yep that looks good